tests/loader/test_olsson_loader.py

```diff
@@ -0,0 +1,152 @@
+        """Tests that the correct pose is fetched (present on disk)."""
+        fetched_pose = self.loader.get_camera_pose(1)
+
+        wRi_expected = np.array(
+            [
+                [0.998079, 0.015881, 0.0598844],
+                [-0.0161175, 0.999864, 0.00346851],
+                [-0.0598212, -0.00442703, 0.998199],
+            ]
+        )
+        wti_expected = np.array([-0.826311, -0.00409053, 0.111315])
+
+        expected_pose = Pose3(Rot3(wRi_expected), wti_expected)
+        self.assertTrue(expected_pose.equals(fetched_pose, 1e-2))
+
+    def test_get_camera_pose_missing(self):
+        """Tests that the camera pose is None, because it is missing on disk."""
+        loader = OlssonLoader(str(NO_EXTRINSICS_FOLDER), image_extension="JPG")
+        fetched_pose = loader.get_camera_pose(5)
+        self.assertIsNone(fetched_pose)
+
+    def test_get_camera_intrinsics_explicit(self):
+        """Tests getter for intrinsics when explicit data.mat file with intrinsics are present on disk."""
+        expected_fx = 2398.119
+        expected_fy = 2393.952
+        expected_fx = min(expected_fx, expected_fy)
+
+        expected_px = 628.265
+        expected_py = 932.382
+
+        computed = self.loader.get_camera_intrinsics(5)
+        expected = Cal3Bundler(
+            fx=expected_fx, k1=0, k2=0, u0=expected_px, v0=expected_py
+        )
+
+        self.assertTrue(expected.equals(computed, 1e-3))
+
+    def test_get_camera_intrinsics_exif(self):
+        """Tests getter for intrinsics when explicit numpy arrays are absent and we fall back on exif."""
+        loader = OlssonLoader(
+            EXIF_FOLDER, image_extension="JPG", use_gt_intrinsics=False
+        )
+        computed = loader.get_camera_intrinsics(5)
+        expected = Cal3Bundler(fx=2378.983, k1=0, k2=0, u0=648.0, v0=968.0)
+        self.assertTrue(expected.equals(computed, 1e-3))
+
+    def test_get_camera_intrinsics_missing(self):
+        """Tests getter for intrinsics when explicit numpy arrays are absent and we fall back on exif."""
+        loader = OlssonLoader(NO_EXIF_FOLDER, image_extension="JPG")
+        computed = loader.get_camera_intrinsics(5)
+        self.assertIsNone(computed)
+
+    def test_create_computation_graph_for_images(self):
+        """Tests the graph for loading all the images."""
+        image_graph = self.loader.create_computation_graph_for_images()
+
+        # check the length of the graph
+        self.assertEqual(12, len(image_graph))
+        results = dask.compute(image_graph)[0]
+
+        # randomly check image loads from a few indices
+        np.testing.assert_allclose(
+            results[5].value_array, self.loader.get_image(5).value_array
+        )
+        np.testing.assert_allclose(
+            results[7].value_array, self.loader.get_image(7).value_array
+        )
+
+    def test_create_computation_graph_for_intrinsics(self):
+        """Tests the graph for all intrinsics."""
```

Comment:
Not sure what is being tested in these two unit tests. Arent tests for the graph supposed to be in the base class?